saagieapi/pipelines/pipelines.py

```diff
@@ -391,20 +411,33 @@ def run_with_callback(self, pipeline_id: str, freq: int = 10, timeout: int = -1)
         res = self.run(pipeline_id)
         pipeline_instance_id = res.get("runPipeline").get("id")
 
-        pipeline_instance_info = self.get_instance(pipeline_instance_id)
+        pipeline_instance_info = self.get_instance(pipeline_instance_id, pprint_result=False)
         state = pipeline_instance_info.get("pipelineInstance").get("status")
 
         sec = 0
         final_status_list = ["SUCCEEDED", "FAILED", "KILLED"]
+
+        logging.info("⏳ Pipeline id %s with instance %s has just been requested", pipeline_id, pipeline_instance_id)
+
         while state not in final_status_list:
-            to = False if timeout == -1 else sec >= timeout
-            if to:
-                raise TimeoutError("Last state known : " + state)
-            time.sleep(freq)
-            sec += freq
-            pipeline_instance_info = self.get_instance(pipeline_instance_id)
-            state = pipeline_instance_info.get("pipelineInstance").get("status")
-            logging.info("Current state : %s", state)
+            with console.status(f"Job is currently {state}", refresh_per_second=100):
+                to = False if timeout == -1 else sec >= timeout
+                if to:
+                    raise TimeoutError("Last state known : " + state)
```

Comment:
Should add fstring and emoji for l568, like:
raise TimeoutError(f"x Last state known : {state}")